libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -689,13 +690,31 @@ void Compass::_detect_backends(void)
 #endif
 
 #if HAL_WITH_UAVCAN
-    if ((AP_BoardConfig::get_can_enable() != 0) && (hal.can_mgr != nullptr))
-    {
-        if((_backend_count < COMPASS_MAX_BACKEND) && (_compass_count < COMPASS_MAX_INSTANCES))
-        {
-            printf("Creating AP_Compass_UAVCAN\n\r");
-            _backends[_backend_count] = new AP_Compass_UAVCAN(*this);
-            _backend_count++;
+    if (AP_BoardConfig_CAN::get_can_enable() != 0) {
+        hal.scheduler->delay(2000);
```

Comment:
Did not understand. We need to wait here for single message to come from external magnetometer. And after that if there was a message - it will get registered if there is a free space for it. Limitation of maximum number of magnetometers and inability to turn off internal ones is a different story and I don't think that CAN mag has automatically higher priority than internal.